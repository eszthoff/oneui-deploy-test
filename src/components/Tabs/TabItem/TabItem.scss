.TabItem {
    background-color: var(--color-neutral-25);
    border: {
        color: var(--color-neutral);
        style: solid;
        top-left-radius: var(--border-radius);
        top-right-radius: var(--border-radius);
        width: var(--line-normal);
        bottom-width: 0;
    }
    color: var(--link-color-normal);
    cursor: pointer;
    font: {
        family: var(--font-family-primary);
        size: var(--font-size-normal);
    }
    margin: {
        top: var(--margin);
        right: var(--margin);
        left: var(--margin);
    };
    outline: none;
    padding: var(--spacing-2x) var(--spacing-4x);
    line-height: var(--line-height-normal);
    text-decoration: none;
    transition: border-color var(--transition-duration);

    &:hover {
        border-color: var(--color-neutral-60);
    }

    &--isActive {
        background-color: var(--color-background);
        color: var(--color-foreground);
        margin-bottom: -1px;

        &:hover {
            border-color: var(--color-neutral);
        }
    }
}
