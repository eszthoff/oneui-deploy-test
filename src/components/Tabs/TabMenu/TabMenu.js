import React from 'react';
import PropTypes from 'prop-types';
import bem from 'bem';
import styles from './TabMenu.scss';

const { block } = bem({
    name: 'TabMenu',
    classnames: styles,
    propsToMods: []
});

const TabMenu = props => {
    const { children, ...rest } = props;
    return (
        <div {...rest} {...block(props)}>
            {children}
        </div>
    );
};

TabMenu.propTypes = {
    children: PropTypes.oneOfType([PropTypes.node, PropTypes.arrayOf(PropTypes.node)])
};

TabMenu.defaultProps = {
    children: null
};

export default TabMenu;
