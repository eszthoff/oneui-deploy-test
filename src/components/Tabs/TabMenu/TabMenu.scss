.TabMenu {
    border-bottom: {
        color: var(--color-neutral);
        style: solid;
        width: var(--line-normal);
    }
    display: flex;
    justify-content: flex-start;
    padding: {
        top: var(--spacing-2x);
        left: var(--spacing-2x);
        right: var(--spacing-2x);
    }
}
