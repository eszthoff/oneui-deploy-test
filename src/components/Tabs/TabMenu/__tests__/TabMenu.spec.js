import React from 'react';
import toJson from 'enzyme-to-json';
import TabMenu from '../TabMenu';
import TabItem from '../../TabItem';

describe('<TabMenu>', () => {
    it('should render tab menu wrapper correctly', () => {
        const wrapper = shallow(
            <TabMenu>
                <TabItem label="Some tab" isActive />
                <TabItem label="Another tab" />
            </TabMenu>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
