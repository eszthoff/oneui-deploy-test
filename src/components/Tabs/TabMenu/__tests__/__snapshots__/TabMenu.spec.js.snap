// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabMenu> should render tab menu wrapper correctly 1`] = `
<div
  className="TabMenu"
>
  <TabItem
    isActive={true}
    label="Some tab"
  />
  <TabItem
    isActive={false}
    label="Another tab"
  />
</div>
`;
